pkg/artifactory/datasource/repository/virtual/datasource_artifactory_virtual_repository_test.go

```diff
@@ -0,0 +1,320 @@
+package remote_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/jfrog/terraform-provider-artifactory/v7/pkg/acctest"
+	"github.com/jfrog/terraform-provider-artifactory/v7/pkg/artifactory/resource/repository"
+	"github.com/jfrog/terraform-provider-artifactory/v7/pkg/artifactory/resource/repository/virtual"
+	"github.com/jfrog/terraform-provider-artifactory/v7/pkg/artifactory/resource/security"
+	"github.com/jfrog/terraform-provider-shared/test"
+	"github.com/jfrog/terraform-provider-shared/util"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+func TestAccDataSourceVirtualAllGenericLikePackageTypes(t *testing.T) {
+	for _, repoType := range virtual.PackageTypesLikeGeneric {
+		// TODO: workaround due to bug in Artifactory 7.55.2, 'bypass_head_requests' inconsistency for terraform repo type.
+		if repoType != "terraform" {
```

Comment:
Are we missing a test for the `terraform` package type?

Reply:
Yes, we are. We are also missing it for the virtual repo resource, because of the bug in RT. I'll try to figure out how to test it.

Reply:
The bug was for Terraform remote repository, not virtual. But we're creating a remote one for the test. `bypass_head_requests` default value is `false`, but for the terraform package type it's automatically set to `true` after the repo was created. In UI it's `false`
I'll set it to `true` for this test because it's not relevant to what we need to test.